modules/presets/index.js

```diff
@@ -225,20 +222,33 @@ export function presetIndex() {
 
       for (let i = 0; i < keyMatches.length; i++) {
         const candidate = keyMatches[i];
-
-        // discard candidate preset if location is not valid at `loc`
-        if (validLocations && candidate.locationSetID) {
-          if (!validLocations[candidate.locationSetID]) continue;
-        }
-
         const score = candidate.matchScore(tags);
+        if (score === -1){
+          continue;
+        }
+        matchCandidates.push({score, candidate});
         if (score > best) {
           best = score;
           match = candidate;
         }
       }
     }
 
+    if (match && match.locationSetID && Array.isArray(loc)){
```

Comment:
```suggestion
    if (match && match.locationSetID !== '+[Q2]' && Array.isArray(loc)){
```

I think this statement might speed it up a bit more.. It means that we can skip this block that does the location checking if the best match is already a global preset.

Reply:
Correct, thanks.

Reply:
Can you benchmark it again to see whether this suggestion did anything?

Reply:
Sure. It knocked down ~300ms in 60s benchmark (I used the one described in the PR) - imo its worth getting every part of second if possible.
After looking into some stats more carefully I've observed and edited above, that the locationsAt call takes <1s in 60s so I don't think that further improving this part would be impactful.